app/models/role.rb

```diff
@@ -319,7 +321,7 @@ def filter_for_permission_add(resource_type, current_filters, search)
   end
 
   def filter_for_permissions_remove(resource_type, current_filters)
-    filter_record = find_filter resource_type, current_filters, nil
```

Comment:
Passing a nil here led the `find_filter` function to only match filters with no search, causing an error when the only matching filter has a search condition.